### DISCUSSION TITLE
Access on_click value from button

### DISCUSSION THREAD

**Discussion by HealthyPear** [zauberzeug/nicegui#3770] [Q&A]
### Question

let's assume `some_function` returns an object,

```python
ui.button("bla", on_click=lambda: some_function())
```

given this button how to I access said object?

this element does not have a `bind_value` method

**Comment by falkoschindler** [comment]
The question is, _when_ do you want to access the result.

If you want to use it right away, you can do so:
```py
ui.button("bla", on_click=lambda: ui.notify(some_function()))
```

If you want to use it later, you need to store the result in some kind of data model, e.g. a simple variable or some larger data structure:
```py
def some_function():
    return 'Hello from Python!'


data = {'message': 'No message yet'}

ui.button('Generate message', on_click=lambda: data.update(message=some_function()))
ui.button('Show message', on_click=lambda: ui.notify(data['message']))
```
